packages/client/lib/AccessTokenClient/IssuanceInitiationAccessTokenClient.ts

```diff
@@ -13,38 +13,55 @@ import {
 import { ObjectUtils } from '@sphereon/ssi-types';
 import Debug from 'debug';
 
-import { MetadataClient } from './MetadataClient';
-import { convertJsonToURI, formPost } from './functions';
+import { IssuanceInitiationClient } from '../CredentialOffer';
+import { MetadataClient } from '../MetadataClient';
+import { convertJsonToURI, formPost } from '../functions';
+
+import { AccessTokenClient } from './AccessTokenClient';
 
 const debug = Debug('sphereon:openid4vci:token');
 
-export class AccessTokenClient {
-  public async acquireAccessTokenUsingIssuanceInitiation({
-    issuanceInitiation,
-    asOpts,
-    pin,
-    codeVerifier,
-    code,
-    redirectUri,
-    metadata,
-  }: AccessTokenRequestOpts): Promise<OpenIDResponse<AccessTokenResponse>> {
-    const { issuanceInitiationRequest } = issuanceInitiation;
+export class IssuanceInitiationAccessTokenClient implements AccessTokenClient {
+  getAccessTokenRequest = (
+    issuanceInitiationClient: IssuanceInitiationClient,
+    metadata: EndpointMetadata,
+    pin?: string,
+    clientId?: string,
+    codeVerifier?: string,
+    code?: string,
+    redirectUri?: string
+  ): IssuanceInitiationAccessTokenRequestOpts => {
+    return {
+      issuanceInitiation: issuanceInitiationClient.issuanceInitiationWithBaseUrl,
+      metadata: metadata,
+      pin,
+      codeVerifier,
+      code,
+      redirectUri,
+      asOpts: { clientId: clientId },
+    };
+  };
 
-    const isPinRequired = this.isPinRequiredValue(issuanceInitiationRequest);
-    const issuerOpts = { issuer: issuanceInitiationRequest.issuer };
+  public async acquireAccessToken(accessTokenRequestOpts: IssuanceInitiationAccessTokenRequestOpts): Promise<OpenIDResponse<AccessTokenResponse>> {
+    const accessTokenRequestOptions = accessTokenRequestOpts as IssuanceInitiationAccessTokenRequestOpts;
```

Comment:
Why is this cast necessary? Also why is there a need for a copy when the param is just being used to read the options?

Reply:
fixed